### PR TITLE
[IMP] mail: move chatter handler inside the chatter model

### PR DESCRIPTION
**Pull request by phenix-factory**
Part of task-2579306

### CODE REVIEW

**Comment by phenix-factory** [comment]
@seb-odoo This one is a bit bigger, but the whole thing is pretty straightforward.

File: addons/mail/static/src/models/chatter/chatter.js [hunk 1]
```diff
@@ -75,6 +176,19 @@ function factory(dependencies) {
             this.focus();
         }
 
+        async refresh() {
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:179]: `refresh()` method was there twice

File: addons/mail/static/src/models/chatter/chatter.js [hunk 2]
```diff
@@ -50,12 +57,106 @@ function factory(dependencies) {
...
+         */
+        onClickAttachment() {
+            this.update({ isAttachmentBoxVisible: !this.isAttachmentBoxVisible });
+        }
+
+
+        /**
+         * Handle message posted even.
+         */
+        onComposerMessagePosted() {
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:84]: and also `onComposerMessagePosted()`

File: addons/mail/static/src/models/chatter/chatter.js [hunk 3]
```diff
@@ -50,12 +57,106 @@ function factory(dependencies) {
...
+
+        /**
+         * Handle click on activity box title
          */
         onClickActivityBoxTitle() {
             this.update({ isActivityBoxVisible: !this.isActivityBoxVisible });
         }
 
+        /**
+         * Handle click on attachement.
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:74]: handle -> handles (everywhere)

File: addons/mail/static/src/models/chatter/chatter.js [hunk 4]
```diff
@@ -50,12 +57,106 @@ function factory(dependencies) {
...
+            }
+            if (this.isComposerVisible && this.composer.isLog) {
+                this.update({ isComposerVisible: false });
+            } else {
+                this.showLogNote();
+            }
+        }
+
+        /**
+         * Handle onCLick on Send message button.
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:103]: click not onCLick

File: addons/mail/static/src/components/chatter_topbar/chatter_topbar.js [hunk 5]
```diff
@@ -1,100 +1,29 @@
...
-    // Public
-    //--------------------------------------------------------------------------
-
     /**
-     * @returns {mail.chatter}
+     * @override
      */
-    get chatter() {
-        return this.messaging && this.messaging.models['mail.chatter'].get(this.props.chatterLocalId);
+    constructor(...args) {
```
Comment (seb-odoo) [inline addons/mail/static/src/components/chatter_topbar/chatter_topbar.js:13]: setup

File: addons/mail/static/src/models/chatter/chatter.js [hunk 6]
```diff
@@ -50,12 +57,106 @@ function factory(dependencies) {
...
+                options: {
+                    on_close: () => {
+                        this.componentChatterTopbar.trigger('reload', { keepChanges: true });
+                    },
+                },
+            });
+        }
+
+        /**
+         * @param {MouseEvent} ev
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:151]: scroll event is not `MouseEvent`, just `Event`

**Review by seb-odoo — COMMENTED** [review]
Few remarks, I'll just push them together with few other docstring fixes and some renames.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**